code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -684,3 +691,43 @@
 
 /mob/living/simple_animal/proc/stop_deadchat_plays()
 	stop_automated_movement = FALSE
+
+
+//Makes this mob hunt the prey, be it living or an object. Will kill living creatures, and delete objects.
+/mob/living/simple_animal/proc/hunt(hunted)
+	if(src == hunted) //Make sure it doesn't eat itself. While not likely to ever happen, might as well check just in case.
+		return
+	if(!((src.loc) && isturf(src.loc))) // Are we on a proper turf & location?
+		stop_automated_movement = FALSE
+		return
+	if(stat || resting || buckled) // Are we concious, upright, and not buckled?
+		stop_automated_movement = FALSE
+		return
+	if(Adjacent(hunted) && COOLDOWN_FINISHED(src, emote_cooldown))
```

Comment:
this is a lot better but you can still get rid of the pyramid'ing by making both of the if statements in this proc that have else statements into early returns.

Reply:
Looking at it again, the only `else` statement I can actually do that with is the one that determines if it's too far away from the object. The rest of the code already returns early at multiple points, and the other `else` statement is needed because it determines if the thing it is hunting is a living being or not.

Reply:
hope that's a good enough amount